### ISSUE TITLE
Fix parse error for nested selectors within property sets in selector-max-*

### ISSUE THREAD

**Issue by jarodtaylor** [stylelint/stylelint#3600]
> Clearly describe the bug

There is an issue when using nested selectors within property sets. This is related to another [closed/unresolved issue](https://github.com/stylelint/stylelint/issues/2783).

> Which rule, if any, is the bug related to?

`selector-max-id`

> What CSS is needed to reproduce the bug?

```
:root {
  --foo: {
    & .bar { color: blue; }
  }
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-max-id": 0
  }
}
```

> Which version of stylelint are you using?

`9.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

stylelint-webpack-plugin (version: 0.10.5)

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to `postcss-apply`.

> What did you expect to happen?

No warning/errors. 

> What actually happened (e.g. what warnings or errors did you get)?

`✖  Cannot parse selector   parseError`

**Comment by jarodtaylor** [comment]
I'm also unable to get stylelint-disable to skip it:

```
 
:root {
  /* stylelint-disable */
  --foo: {
    & .bar { color: blue; }
  }
 /* stylelint-enable */
}
 
```

I still get the same `parseError`.

**Comment by jeddy3** [comment]
@jarodtaylor Thanks for the report and for using the template.

The other issue was closed by the original reporter due to inactivity, but the bug is still there. I mentioned in https://github.com/stylelint/stylelint/issues/2783#issuecomment-372362265 that the fix should be pretty simple and involves returning early if the _resolved selector_ is non-standard e.g. putting something [like this](https://github.com/stylelint/stylelint/blob/35a24b8ef3a481ed9fa4dfa37d9f4cef6b5dda51/lib/rules/selector-max-specificity/index.js#L123-L126) between [lines 68 and 69](https://github.com/stylelint/stylelint/blob/69b6de50bcebc1c695e7e9ffd530fa39beeb71f3/lib/rules/selector-max-id/index.js#L68-L69).

As the issue has been labelled "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) to this rule (and the other `selector-max-* ones).

> I'm also unable to get stylelint-disable to skip it:

I think this is because the rule is still attempting to parse the code as the disable comments remove violations from the result _after the fact_. I think once the rules are patched up, this behaviour is unlikely to come up again. If it does, though, we can always address it then.

**Comment by jarodtaylor** [comment]
Thanks, @jeddy3. I may try to get this fix in, this evening, and put in a PR.

**Comment by jarodtaylor** [comment]
@jeddy3 I'm not sure the `isStandardSyntaxSelector` returns in this instance. I've added the following  to the `selector-max-id` test file in the `accept` array:

```js
{
  code: ":root { --custom-property-set: { & .foo {} } }",
  description: "nested selectors in custom property set in root"
},
```

So, as expected this throws a test failure. When I add this:
```js
//Skip non-standard syntax selectors
if (!isStandardSyntaxSelector(resolvedSelector)) {
  return;
}
```

There is no change. So, I tested this by putting the same code/description inside of the `selector-max-specificity` test file. I get the same error, because I don't think `isStandardSyntaxSelector` is ever true with that particular syntax.

**Comment by jeddy3** [comment]
> is ever true with that particular syntax.

I think you're right. I was thinking of the custom property set conditonal in [`isStandardSyntaxRule`](https://github.com/stylelint/stylelint/blob/1ff68f90a228f1d2d6319c44ad07d168910259ed/lib/utils/isStandardSyntaxRule.js#L14-L17). I don't think we can make use of this util here, so we'll probably need to add a heuristic to `isStandardSyntaxSelector` to check for this particular syntax. Maybe we can check for selectors that contain a selector that starts with `--` and ends with `:`?

**Comment by jarodtaylor** [comment]
@jeddy3 yep, that’s what I was looking at. I’ll take a stab at it, today.

**Comment by ybiquitous** [comment]
Does this error still occur? 🤔 
I check two reproduction code snippets on the demo, but I cannot see the error:

- [snippet using `:root`](https://stylelint.io/demo/#N4Igxg9gJgpiBcCBOEIBcAEwA6A7DGAtIQGarxZ4EEBkGAdAEYCGSWGkANhEhY5wFcYAbgwBfKuLxiQAGhABnAJ640zAB4JFYBQrngIuEgEsA5lpz4M2EEgGcYCmxUvVrimA7BoehALYahMZQzhgADJISuDJiQA)
- [snippet using `map-deep-get`](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0BnCBbGBaALgQwCMAbGAfVICcIB3eAOxwAssxGBLIqACmigEp4eAGY4YFAATAAOnXFyCeMAGsA5lQCudKCwhEIFeOPR4ADllgwzKmDi4ASNJlyESAGnELlaiJv4BuGQBfEFcQFABPBjwADwQwsBQUEPAIOiE2FTjpWXEpEAp1EhQ8w2y5OTyUGBIwHH0sY2isNigS8QAGGTlAoJBAoA)
